NMS-14614: Quick Start - Notifications (#5204)
OpenNMS/opennms: docs/modules/operation/pages/quick-start/users.adoc

The default user login is `admin` with the password `admin`.

The first time you log in, we prompt for permission to allow the Data Choices module to collect and publish anonymous usage statistics to https://stats.opennms.org.
The OpenNMS Group uses this anonymized information to help determine product usage and to improve the {page-component-title} software.
We do not share it with third parties.

image::users/data-sources.png[prompt,300]

> Click *Show me What is Being Sent* to see what xref:deep-dive/admin/housekeeping/introduction.adoc#ga-data-collection[information we collect].

Reviewer: ```suggestion
Click *Show me what is being sent* to see what xref:deep-dive/admin/housekeeping/introduction.adoc#ga-data-collection[information we collect].
```
We don't need to reproduce the exact capitalization on screen, especially when it's wrong. :-)

Author: Sounds good. I do seem to remember something about using title case for buttons in the Style Guide, but I can't find it right now. :p

After logging in for the first time, make sure to change the default `admin` account password to a secure one:

. Click the *Gear* icon at the top-right of the screen.
. Choose menu:OpenNMS System[Configure Users, Groups and On-Call Roles, Configure Users].
. Click *Modify* beside the `admin` user.
. In the User Password area, click *Reset Password*.
. Update the password and click *OK*.
. Click *Finish* at the bottom of the *Modify User* screen to save changes.

We recommend that you do not use the default `admin` user account, but instead xref:deep-dive/user-management/user-config.adoc#ga-user-config [create specific users] with the Admin role and other permissions.
This helps to keep track of who has performed tasks such as clearing alarms or creating notifications.

> You should still change the password of the `admin` user to be more secure.

Author: ```suggestion

For the purposes of this Quick Start guide, create one new user account for yourself and assign it the Admin role.
Include your email address and any other relevant contact information for notification routing:

. Click the *Gear* symbol at the top-right of the screen.
. Choose menu:OpenNMS System[Configure Users, Groups, and On-Call Roles, Configure Users].
. Click *Modify* beside your user account.
. In the Notification Information section, add your email address.
. (Optional) Add other contact information (for example, a mobile phone number).
. Click *Finish* at the bottom of the screen to save your changes.

You will need this user account for the next page: <<quick-start/notifications.adoc, Step 2: Turn on Default Notifications>>.
```

WARNING: Do not delete the default `admin` and `rtc` user accounts.
The `rtc` user account is used for the communication of the Real-Time Console on the start page to calculate the node and service availability.
You should still change the password of the `admin` user to be more secure.

[[ga-users-next]]
== Next steps

> Before xref:operation:quick-start/notifications.adoc[setting up default notifications], you must configure notification information for any users with admin privileges:

Author: ```suggestion
Before xref:operation:quick-start/notifications.adoc[setting up default notifications], make sure you have configured contact information for your account.
At minimum, your user account must have an email address associated with it to receive default notifications when they are turned on.
```

After logging in for the first time, make sure to change the default `admin` account password to a secure one:

. Click the *Gear* icon at the top-right of the screen.
. Choose menu:OpenNMS System[Configure Users, Groups and On-Call Roles, Configure Users].
. Click *Modify* beside the `admin` user.
. In the User Password area, click *Reset Password*.
. Update the password and click *OK*.
. Click *Finish* at the bottom of the *Modify User* screen to save changes.

We recommend that you do not use the default `admin` user account, but instead xref:deep-dive/user-management/user-config.adoc#ga-user-config[create specific users] with the Admin role and other permissions.
This helps to keep track of who has performed tasks such as clearing alarms or creating notifications.

> For the purposes of this Quick Start guide, create one new user account for yourself and assign it the Admin role.

Author: ```suggestion
For the purposes of this Quick Start guide, xref:operation:deep-dive/user-management/user-config.adoc#ga-user-create[create one new user account] for yourself and assign it the Admin role.
```

Include your email address and any other relevant contact information for notification routing:

> 

Reviewer: Since we're including the configuration, let's just have them create the user here.

Author: This may be a more significant change, so I'll do it locally and push it. :)

. Click the *Gear* symbol at the top-right of the screen.
. Choose menu:OpenNMS System[Configure Users, Groups, and On-Call Roles, Configure Users].

> . Click *Modify* beside your user account.

Reviewer: ```suggestion
. Click *Add new user* and specify a user ID and password for yourself, confirm the password, and click *OK*.
```

. In the Notification Information section, add your email address.
. (Optional) Add other contact information (for example, a mobile phone number).
. Click *Finish* at the bottom of the screen to save your changes.

> You will need this user account for the next page: <<quick-start/notifications.adoc, Step 2: Turn on Default Notifications>>.

Reviewer: ```suggestion
You will need this user account for the next step: <<quick-start/notifications.adoc, Turn on Default Notifications>>.
```

. Click the *Gear* symbol at the top-right of the screen.
. Choose menu:OpenNMS System[Configure Users, Groups and On-Call Roles > Configure Users].
. Click *Modify* beside the `admin` user.
. In the User Password area, click *Reset Password*.
. Update the password and click *OK*.
. Click *Finish* at the bottom of the *Modify User* screen to save changes.

We recommend that you do not use the default `admin` user account, but instead xref:deep-dive/user-management/user-config.adoc#ga-user-config[create specific users] with the Admin role and other permissions.
This helps to keep track of who has performed tasks such as clearing alarms or creating notifications.

For the purposes of this Quick Start guide, create one new user account for yourself, assign it the Admin role, and configure your email address and any other relevant contact information:

. Log in using the default `admin` account.
. Click the *Gear* symbol at the top-right of the screen, and choose menu:Configure OpenNMS[Configure Users, Groups and On-Call Roles > Configure Users].
. Click *Add New User*, specify a user ID and password, confirm the password, and click *OK*.
. Enter your information:

> ** In the User Information section, type your name in the Full Name box.

Reviewer: ```suggestion
** In the User Information section, type your name in the *Full Name* box.
```

> ** In the Security Roles section, select `ROLE_ADMIN` in the Available Roles list and click *Add >>* to add it to your account.

Reviewer: ```suggestion
** In the Security Roles section, select `ROLE_ADMIN` in the *Available Roles* list and click *Add >>* to add it to your account.
```

*** (Optional) Add xref:operation:deep-dive/user-management/security-roles.adoc[other security roles].

> ** In the Notification Information section, add your email address in the Email box.

Reviewer: ```suggestion
** In the Notification Information section, add your email address in the *Email* box.
```

*** (Optional) Add other contact information (for example, a mobile phone number).
. Click *Finish* at the bottom of the screen to create the account.

You will need this user account for the next step: xref:operation:quick-start/notifications.adoc[Turn on Default Notifications].

WARNING: Do not delete the default `admin` and `rtc` user accounts.
The `rtc` user account is used for the communication of the Real-Time Console on the start page to calculate the node and service availability.
You should still change the password of the `admin` user to be more secure.

[[ga-users-next]]

> == Next steps

Reviewer: We can remove the next steps now that we have the procedure included above.

> 
> [[ga-users-next]]
> == Next steps
> 
> Before xref:operation:quick-start/notifications.adoc[setting up default notifications], make sure you have configured contact information for your account.
> At minimum, your user account must have an email address associated with it to receive default notifications when they are turned on.

Author: ```suggestion
```

. Update the password and click *OK*.
. Click *Finish* at the bottom of the *Modify User* screen to save changes.

> We recommend that you do not use the default `admin` user account, but instead xref:deep-dive/user-management/user-config.adoc#ga-user-config [create specific users] with the Admin role and other permissions.

Reviewer: ```suggestion
We recommend that you do not use the default `admin` user account, but instead xref:deep-dive/user-management/user-config.adoc#ga-user-config [create specific users] with the Admin role and other permissions.
For the purpose of the quick start, create one user for yourself, with admin permissions, and include your email address and any other contact information (for example, a mobile phone number).
You will use this user in <<quick-start/notifications.adoc, Step 2: Turn on Default Notifications>> of the quick start.
```

Author: I'm going to move the extra content here down, so that it doesn't break the flow of recommendations for the `admin` account.

[[ga-users-next]]
== Next steps

> Before xref:operation:quick-start/notifications.adoc[setting up default notifications], you must configure notification information for any users with admin privileges:

Reviewer: Because this is a quick start, so far they have only one user, the Admin user, so there aren't any other users to modify. While we referred them to the create other users section, we really only want them to create one user at this point. 
Incorporated this content above or we can update the current procedure below to be "Create a new user".
